app/actions/remote/post.ts

```diff
@@ -552,3 +552,45 @@ export const togglePinPost = async (serverUrl: string, postId: string) => {
         return {error};
     }
 };
+
+export const markPostAsUnread = async (serverUrl: string, postId: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+    let client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        const [userId, post] = await Promise.all([queryCurrentUserId(database), queryPostById(database, postId)]);
+        if (post && userId) {
+            //marks the post as unread on the server
+            client.markPostAsUnread(userId, postId);
```

Comment:
we should await to ensure that the next set of requests can retrieve the right data and avoid race conditions

Reply:
done